commands/miscellaneous/getreactions.js

```diff
@@ -1,47 +1,39 @@
+const { SlashCommandBuilder } = require('@discordjs/builders');
 
-module.exports = class RemoveRoleCommand extends Command {
-  constructor(client) {
-    super(client, {
-      name: "reactions",
-      aliases: ["reactions"],
-      group: "miscellaneous",
-      memberName: "get-reactions",
-      description: "",
-      userPermissions: ['MANAGE_CHANNELS', 'MANAGE_ROLES'],
-      args: [
-        {
-          key: "messageID",
-          prompt: "What message would you like to get reactions from?",
-          type: "string",
-        },
-        {
-            key: "channel",
-            prompt: "In what channel is this message?",
-            type: "channel",
-          },
-      ],
-    });
-  }
+// TODO: needs permissions 'MANAGE_CHANNELS', 'MANAGE_ROLES'
+module.exports = {
+    data: new SlashCommandBuilder()
+        .setName('getreactions')
+        .setDescription('Get reactions from a message')
+        .addStringOption(option =>
+            option.setName('message_id')
+                .setDescription('The message id you would like to get reactions from')
+                .setRequired(true))
+        .addChannelOption(option =>
+            option.setName('channel')
+                .setDescription('The channel this message is in')
+                .setRequired(true)),
+    async execute(interaction) {
+        const options = interaction.options;
+        const messageId = options.getString('message_id');
+        const channel = options.getChannel('channel');
 
-  run(message, { messageID, channel }) {
-
-    channel.messages.fetch(messageID).then( ReactionsMessage => {
-        var users = {};
-        ReactionsMessage.reactions.cache.forEach(reaction => {
-            users[reaction.emoji] = []
-            reaction.users.fetch().then( result => {
-                result.forEach( user => {
-                    users[reaction.emoji].push("<@" + user.id + ">");
-                })
-                var attachment = new Discord.MessageAttachment(Buffer.from(`${users[reaction.emoji].join('\n')}`, 'utf-8'), 'emoji reactions.txt');
-                message.channel.send(`Users that reacted with ${reaction.emoji}`, attachment);
+        channel.messages.fetch(messageId).then((ReactionsMessage) => {
+            const users = {};
+            ReactionsMessage.reactions.cache.forEach((reaction) => {
+                users[reaction.emoji] = [];
+                reaction.users.fetch().then(result => {
+                    result.forEach(user => {
+                        users[reaction.emoji].push('<@' + user.id + '>');
+                    });
+                    const attachment = new Discord.MessageAttachment(Buffer.from(`${users[reaction.emoji].join('\n')}`, 'utf-8'), 'emoji reactions.txt');
+                    interaction.channel.send({ content: `Users that reacted with ${reaction.emoji}`, files: [attachment] });
+                });
             });
+        }).catch((error) => {
+            console.log(error);
```

Comment:
Probably a question for the discord channel -- does this actually get logged to somewhere we can see / reference when an error occurs (not in a testing environment)? If not, we should leave the console logs out of the main code.

Reply:
The error gets save in a err.log fyi. 😊 Natalie and I can refer to it if there's any issues with the commands.